### ISSUE TITLE
Generate a Progressive Web App

### ISSUE THREAD

**Issue by marshallswain** [donejs/generator-donejs#251]
Just for kicks, and inspired in part by [Matthew's HTTP 2 article](https://www.bitovi.com/blog/utilizing-http2-push-in-a-single-page-application), I tried running our default `donejs add app` code through the [Chrome Lighthouse extension](https://developers.google.com/web/tools/lighthouse/) that rates PWAs.  

The full checklist for creating a PWA is [here](https://developers.google.com/web/tools/lighthouse/).

Below is the output from the plugin.  If we don't have anybody working on getting the default score higher, I'd like to start contributing towards that goal.

![screen shot 2017-06-05 at 8 03 33 pm](https://cloud.githubusercontent.com/assets/128857/26810738/9ba29c18-4a2a-11e7-9df8-394cf02e7974.jpg)
![screen shot 2017-06-05 at 8 03 54 pm](https://cloud.githubusercontent.com/assets/128857/26810743/9e80b8a2-4a2a-11e7-8e1b-4a333de7e398.jpg)
![screen shot 2017-06-05 at 8 04 08 pm](https://cloud.githubusercontent.com/assets/128857/26810748/a12b4176-4a2a-11e7-8e46-aec6159f4663.jpg)

**Comment by justinbmeyer** [comment]
Is this dev mode?

**Comment by marshallswain** [comment]
Yes.  The above is in dev mode.  Below is the report after running `npm run build` and `npm run start`.

![screen shot 2017-06-06 at 7 34 20 am](https://cloud.githubusercontent.com/assets/128857/26831792/a207e65c-4a8a-11e7-8ed4-a261b6deb467.jpg)
![screen shot 2017-06-06 at 7 35 13 am](https://cloud.githubusercontent.com/assets/128857/26831828/ba08ec7e-4a8a-11e7-9da1-a54e6e693b58.jpg)

**Comment by phillipskevin** [comment]
@marshallswain Is this a duplicate of https://github.com/donejs/donejs/issues/734 and https://github.com/donejs/donejs/issues/735?

**Comment by marshallswain** [comment]
Almost!  mine has pictures.

**Comment by phillipskevin** [comment]
Awesome. Thanks!
